src/ray/object_manager/object_manager.cc

```diff
@@ -31,7 +31,8 @@ ObjectManager::ObjectManager(asio::io_service &main_service,
                    /*release_delay=*/2 * config_.max_sends),
       send_work_(send_service_),
       receive_work_(receive_service_),
-      connection_pool_() {
+      connection_pool_(),
+      recent_pushes_timer_(main_service) {
```

Comment:
Rather than use a timer, you can store timestamps in the recent push list and ignore expired entries lazily.

Reply:
Btw, it is possible to use a library structure like a lru cache which would be O(1) to check? https://www.boost.org/doc/libs/1_68_0/boost/compute/detail/lru_cache.hpp

Reply:
I got rid of the timer and am using timestamps now. Lookups are now O(1).